iceoryx_posh/include/iceoryx_posh/internal/popo/building_blocks/condition_variable_data.hpp

```diff
@@ -38,16 +38,11 @@ struct ConditionVariableData
     ConditionVariableData& operator=(ConditionVariableData&& rhs) = delete;
     ~ConditionVariableData() noexcept = default;
 
-    posix::Semaphore m_semaphore = std::move(
-        posix::Semaphore::create(posix::CreateUnnamedSharedMemorySemaphore, 0U)
-            .or_else([](posix::SemaphoreError&) {
-                errorHandler(PoshError::POPO__CONDITION_VARIABLE_DATA_FAILED_TO_CREATE_SEMAPHORE, ErrorLevel::FATAL);
-            })
-            .value());
-
+    cxx::optional<posix::UnnamedSemaphore> semaphore;
```

Comment:
You gonna hate me but this is the same like with `wasNotified`. Currently we don't require to have the `m_` prefix for public member but for consistency reasons it would be good to not mix the style in one struct.